src/subclass/widget.rs

```diff
@@ -814,6 +838,64 @@ impl<T: WidgetImpl + ObjectImpl> WidgetImplExt for T {
             );
         }
     }
+
+    fn parent_realize(&self, widget: &Widget) {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut gtk_sys::GtkWidgetClass;
+            let f = (*parent_class)
+                .realize
+                .expect("No parent class impl for \"realize\"");
+            f(widget.to_glib_none().0)
```

Comment:
```suggestion
            if let Some(f) = (*parent_class).realize {
                f(widget.to_glib_none().0);
            }
```

And the same for unrealize, map, unmap and motion_notify_event. They're all optional and nothing bad (and nothing at all) happens when they are not set.

For the motion_notify_event, see the other _event impls for what to do about the return value